plugins/kubernetes-backend/src/kubernetes-auth-translator/AzureIdentityKubernetesAuthTranslator.ts

```diff
@@ -31,11 +42,23 @@ export class AzureIdentityKubernetesAuthTranslator
       clusterDetails,
     );
 
-    const credentials = new DefaultAzureCredential();
+    if (!this.accessToken || this.tokenExpired()) {
+      this.accessToken = await this.tokenCredential.getToken(aksScope);
```

Comment:
Highlighting that there is a bit of race here where multiple simultaneous token refreshes can happen, although except for the extra requests that's probably fine

Reply:
Yeah, I thought about using some locking mechanism, but in the end it's not going to cause any issues, so I decided to keep it simple.

Reply:
Typical way around this would be to store a reference to the promise instead, but agree that it's not something that's needed or necessarily even worth the complexity